### PR TITLE
fix(classifier): recognize previous_response_id and conversation as Responses signals

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Adds `previous_response_id` and `conversation` as Responses API discriminators in `classify_format`, so bodies without `input` (e.g. `{"model":"gpt-4.1","previous_response_id":"resp_..."}`) are no longer misclassified as `UnknownJson` and rejected with a 400 when `on_invalid: reject` is configured.
- Adds 5 test cases covering the new classification paths, including null-value edge cases and regression guards.

## Test plan

- [x] `cargo test -p praxis-proxy-filter --features ai-inference -- classifier` — all 61 tests pass
- [x] `make lint` — clippy and fmt clean (pre-existing `lint-filter-docs` failure unrelated)
- [x] `make build` — workspace builds successfully

### CODE REVIEW

**Comment by leseb** [comment]
draft because we need https://github.com/praxis-proxy/praxis/pull/693

File: filter/src/builtins/http/ai/classifier/mod.rs [hunk 1]
```diff
@@ -800,6 +805,79 @@ mod tests {
...
+        );
+        assert!(
+            !result.has_conversation,
+            "null value should not set the has_conversation flag"
+        );
+    }
+
     #[test]
     fn control_char_model_extracted() {
         let body = b"{\"model\":\"bad\\nmodel\",\"input\":\"test\"}";
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/classifier/mod.rs:908]: 173ee74402b5064212109bef4c29f159c3bb1de3

**Review by franciscojavierarceo — APPROVED** [review]
